### ISSUE TITLE
[BUG] news feeds will not be updated automatically

### ISSUE THREAD

**Issue by ghost** [FreshRSS/FreshRSS#5989]
Hi,

I have the following error:
After updating to FreshRSS 1.23.0, the news feeds will not be updated automatically.
Hosted on a server (PHP 8.2), I start a php script via cron job:
```

<?php
exec("/usr/bin/php app/actualize_script.php 2>&1", $out, $result);
echo "Return code: " .$result ."<br>";
echo "Script output: " ."<br>";
echo "<pre>"; print_r($out);
?>

```

This should start the php script in the /app/.... directory (the php script is there unchanged from the installation). This worked without any problems up to version 1.22.1.

I use FreshRSS with the “Fiery Feeds” app (iOS 17.2.1), which has also updated without any problems and displayed current feeds.
The feeds can only be updated manually on the server.

**To Reproduce**
1. Update to 1.23.0
2. use Cronjob (like described) to start update-php

**Expected behavior**
As in the previous versions, the php script should update the news feeds every 20 minutes (cron job execution rhythm).

**Environment information (please complete the following information):**
- Device: iPhone 14 Pro
- OS: iOS 17.2.1
- Browser: Safari, Firefox
- FreshRSS version: 1.23.0
- Database version: SQLite
- PHP version: PHP 8.2
- Installation type: all-inkl.com

**Comment by Alkarex** [comment]
Could you please try with our `edge` version?

**Comment by vdbhb59** [comment]
FYI: Works for me on both 1.23.0 as well as 1.23.1-dev

**Comment by ghost** [comment]
i'll give feedback....

**Comment by ghost** [comment]
It is the same, no automatically updates.
I did a fresh install (other webspace), using 1.23.0, Even with this version the automatic updates did not work.

It is still nessessary to update manually in webinterface.

**Comment by Alkarex** [comment]
Please try with our `edge` version

**Comment by vdbhb59** [comment]
> It is the same, no automatically updates. I did a fresh install (other webspace), using 1.23.0, Even with this version the automatic updates did not work.
> 
> It is still nessessary to update manually in webinterface.

Please try 1.23.1-dev (https://github.com/FreshRSS/FreshRSS/archive/edge.zip) and see if it helps?

**Comment by ghost** [comment]
Using now edge on both installations, my „old“ and the formally „clean“ one. Waiting for results….

**Comment by ghost** [comment]
Nothing changed. The Feeds dis no Update via cronjobAm 28.12.2023 um 17:48 schrieb Chris M. ***@***.***>:﻿Using now edge on both installations, my „old“ and the formally „clean“ one. Waiting for results….Am 28.12.2023 um 17:40 schrieb vdbhb59 ***@***.***>:
It is the same, no automatically updates. I did a fresh install (other webspace), using 1.23.0, Even with this version the automatic updates did not work.It is still nessessary to update manually in webinterface.
Please try 1.23.1-dev (https://github.com/FreshRSS/FreshRSS/archive/edge.zip) and see if it helps?—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by ghost** [comment]
Hey,I did a rollback to 1.22.1. The Feeds are updating automatically. After installing the Edge an manually starting the updatescript, i got this Message:Returncode: 0Ausgabe des Scripts: Array
(
    [0] => FreshRSS error: FreshRSS requires PHP 7.4.0+!
)There‘s php 8.2 installed…Am 28.12.2023 um 17:48 schrieb Chris M. ***@***.***>:﻿Using now edge on both installations, my „old“ and the formally „clean“ one. Waiting for results….Am 28.12.2023 um 17:40 schrieb vdbhb59 ***@***.***>:
It is the same, no automatically updates. I did a fresh install (other webspace), using 1.23.0, Even with this version the automatic updates did not work.It is still nessessary to update manually in webinterface.
Please try 1.23.1-dev (https://github.com/FreshRSS/FreshRSS/archive/edge.zip) and see if it helps?—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by ghost** [comment]
Sorry, forgot the script-Message using 1.22.1:Returncode: 0Ausgabe des Scripts: Array
(
    [0] => FreshRSS starting feeds actualization at 2023-12-28T23:05:22+01:00
    [1] => FreshRSS actualize Chrisâ€¦
    [2] => FreshRSS actualize Chris_M.â€¦
    [3] => FreshRSS actualization done for 2 users, using 4.00 MiB of memory, in 0 day(s), 0 hour(s), 0 minute(s) and 0 seconds.
    [4] => Results:
    [5] => Chris OK
    [6] => Chris_M. OK
    [7] => End.
)Am 28.12.2023 um 17:48 schrieb Chris M. ***@***.***>:﻿Using now edge on both installations, my „old“ and the formally „clean“ one. Waiting for results….Am 28.12.2023 um 17:40 schrieb vdbhb59 ***@***.***>:
It is the same, no automatically updates. I did a fresh install (other webspace), using 1.23.0, Even with this version the automatic updates did not work.It is still nessessary to update manually in webinterface.
Please try 1.23.1-dev (https://github.com/FreshRSS/FreshRSS/archive/edge.zip) and see if it helps?—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by Alkarex** [comment]
If you get a message regarding a wrong PHP version, a classic problem is that you might have multiple PHP versions installed, with different versions used by command-line / cron and by your Web server.

Try for instance:

```sh
ls /etc/php/
```

**Comment by ghost** [comment]
My provider only allows you to set the PHP version on the domain and subdomain. All of them are set to 8.2.As a test, I ran one of the two freshrss installations again on 1.22.1 without making any further changes. Everything works perfectly here. the edge has errors.As I said, nothing has been modified in the other settings, cron jobs or anything else.How and where can I search for further causes?Am 29.12.2023 um 00:54 schrieb Alexandre Alapetite ***@***.***>:﻿
If you get a message regarding a wrong PHP version, a classic problem is that you might have multiple PHP versions installed, with different versions used by command-line / cron and by your Web server.
Try for instance:
ls /etc/php/

**Comment by vdbhb59** [comment]
> My provider only allows you to set the PHP version on the domain and subdomain. All of them are set to 8.2.As a test, I ran one of the two freshrss installations again on 1.22.1 without making any further changes. Everything works perfectly here. the edge has errors.As I said, nothing has been modified in the other settings, cron jobs or anything else.How and where can I search for further causes?Am 29.12.2023 um 00:54 schrieb Alexandre Alapetite ***@***.***>:﻿ If you get a message regarding a wrong PHP version, a classic problem is that you might have multiple PHP versions installed, with different versions used by command-line / cron and by your Web server. Try for instance: ls /etc/php/ —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

If using cPanel, try setting the path in php.ini or via php selector.
I use different versions of PHP for my different sub domains as per the need.

**Comment by ghost** [comment]
Set php back to 7.4. Same error:Am 29.12.2023 um 06:30 schrieb vdbhb59 ***@***.***>:﻿

My provider only allows you to set the PHP version on the domain and subdomain. All of them are set to 8.2.As a test, I ran one of the two freshrss installations again on 1.22.1 without making any further changes. Everything works perfectly here. the edge has errors.As I said, nothing has been modified in the other settings, cron jobs or anything else.How and where can I search for further causes?Am 29.12.2023 um 00:54 schrieb Alexandre Alapetite @.>:﻿ If you get a message regarding a wrong PHP version, a classic problem is that you might have multiple PHP versions installed, with different versions used by command-line / cron and by your Web server. Try for instance: ls /etc/php/ —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: @.>

If using cPanel, try setting the path in php.ini or via php selector.
I use different versions of PHP for my different sub domains as per the need.

**Comment by ghost** [comment]
I may have found a solution. In my opinion, this is neither a good nor a practical way. Except that these requirements are not listed in the freshrss documentation.Here is the script that is called from the cron job (with error situation):and here, the way it works now (I have to specify the php version in the path (who is supposed to know, this is more of a chance discovery than knowledge):From my point of view, there is a problem with freshrss's version determination, could that be? As I said, up to and including 1.22.1 the script could be used WITHOUT specifying the php version.Am 29.12.2023 um 06:37 schrieb Chris M. ***@***.***>:﻿Set php back to 7.4. <Login w01d43fb (Hauptaccount).jpeg>Same error:<image.png>Am 29.12.2023 um 06:30 schrieb vdbhb59 ***@***.***>:﻿

My provider only allows you to set the PHP version on the domain and subdomain. All of them are set to 8.2.As a test, I ran one of the two freshrss installations again on 1.22.1 without making any further changes. Everything works perfectly here. the edge has errors.As I said, nothing has been modified in the other settings, cron jobs or anything else.How and where can I search for further causes?Am 29.12.2023 um 00:54 schrieb Alexandre Alapetite @.>:﻿ If you get a message regarding a wrong PHP version, a classic problem is that you might have multiple PHP versions installed, with different versions used by command-line / cron and by your Web server. Try for instance: ls /etc/php/ —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: @.>

If using cPanel, try setting the path in php.ini or via php selector.
I use different versions of PHP for my different sub domains as per the need.

**Comment by Alkarex** [comment]
Could you please write your answers from the Web interface? The old messages are also included when you answer by e-mail, making it a mess to read.

Since FreshRSS 1.23.0 we require PHP 7.4+, and the requirements **are** written.

The error you are facing is that you probably have multiple PHP versions installed, and your default one is too old. But good that you found a solution.

**Comment by vdbhb59** [comment]
I am sad to say this, but now I too am facing this issue since early evening yesterday. I even set it to run as 20 minutes to check if it works, but nothing happens and there are no logs in /data/user/** to provide any detail at all. Very odd! My script is as needed and was running fine up until per 2 hours before I started getting this issue. Some users have also reached out to me stating they are facing this since 2 days (around 5 users have come back with the same thing).

@Alkarex any idea where can I give you logs from, or how do I manage to get some logs?
There is no change at all from my end other than upgrading and running those tests with Alex earlier, so not sure where the issue is now.

The script says:
```
#!/usr/bin/env php
<?php
// declare(strict_types=1);	// Need to wait for PHP 8+ due to https://php.net/ob-implicit-flush
require(__DIR__ . '/../cli/_cli.php');
```

Also, there is this just one error in the log: `Invalid action name “uKuhkaAction” for controller “FreshRSS_index_Controller”.`

**Comment by Alkarex** [comment]
@vdbhb59 `uKuhka` is not something that comes from FreshRSS. Do you have any FreshRSS extension enabled?

**Comment by Alkarex** [comment]
@vdbhb59 Please look also at the output of your cron jobs. 
If you are using the same convention than in [our documentation](https://freshrss.github.io/FreshRSS/en/admins/08_FeedUpdates.html), it could be in `/tmp/FreshRSS.log`

**Comment by vdbhb59** [comment]
> @vdbhb59 `uKuhka` is not something that comes from FreshRSS. Do you have any FreshRSS extension enabled?

I know, and that is why I was confused. Moreover, for some reason, even though I had not turned ON BLACKLIST extension, it was picking up that one. As well as WHITELIST extension. I only had those installed/copied.

I removed those 2 extensions and now this error is not coming.

**Comment by vdbhb59** [comment]
> @vdbhb59 Please look also at the output of your cron jobs. If you are using the same convention than in [our documentation](https://freshrss.github.io/FreshRSS/en/admins/08_FeedUpdates.html), it could be in `/tmp/FreshRSS.log`

Using the normal cron from cPanel as:
`/opt/cpanel/ea-php**/root/usr/bin/php /**/**/public_html/**/feeds/app/actualize_script.php > /**/**/public_html/**/feeds/data/users/FreshRSS_cron.log 2>&1`

& somehow by its own now it is auto refreshing. Not sure what is happening? :( (feeling sad here)...

**Comment by Alkarex** [comment]
The WhiteList extension should be fixed after https://github.com/aledeg/xExtension-WhiteList/pull/2

**Comment by vdbhb59** [comment]
> The WhiteList extension should be fixed after [aledeg/xExtension-WhiteList#2](https://github.com/aledeg/xExtension-WhiteList/pull/2)

Perfect. I will try this out and revert.
